WasatchNET/HOCTSpectrometer.cs

```diff
@@ -0,0 +1,1238 @@
+
+            ushort[] RawPixelData = Enumerable.Repeat((ushort)0, OctUsb.iNumOfPixels).ToArray();
+            int iFramesTransmitted = 1;
+
+            bool bError = false;
+            bool bFirst = true;
+
+            while (!CT.IsCancellationRequested)
+            {
+                lock (acquisitionLock)
+                    RawPixelData = OctUsb.CaptureBitMap(iFramesTransmitted, true, bFirst, ref bError);
+
+                if (bError == false)
+                {
+                    lock (frameLock)
+                        lastFrame = RawPixelData;
+                    bFirst = false;
+                }
+                
+                Thread.Sleep(10);
+            }
+
+            OctUsb.ControlBoardSim(0);
+            OctUsb.DisarmCapture();
+            OctUsb.ClearProcessingBuffer();
+        }
+
+        public override void close()
+        {
+            if (commsOpen)
+            {
+                lock (acquisitionLock)
+                    _cancellationTokenSource.Cancel();
+
+                FrameProcess.Wait();
+
+                bool closeOk = OctUsb.CloseDevice();
+                if (closeOk)
+                    commsOpen = false;
+            }
+        }
+
+        public override double[] getSpectrum(bool forceNew = false)
+        {
+            if (forceNew)
+            {
+                // 11.25 is based on the HOCT clock rate (11.25), and dividing by 5 is for *200 lines per frame and /1000
+                // to convert from us to ms
+                // The line rate is at least 100 us, in practice we've seen returns take up to ~60-70 ms, hard to say where exactly
+                // all the delay comes from. 20 ms is theoretical minimum. We give an extra buffer for long reads then an extra
+                // 10 for the frame loop's sleep
+                //
+                // May reduce the minimum or change the frame vs. usb split with more experimentation
+                int minWait = 20;
+
+                int wait = (int)((integrationTimeMS / 11.25) / 5);
+                if (wait < minWait)
+                    wait = minWait;
+
+                //give time for loop wait and usb read back
+                wait += 70;
+
+                Thread.Sleep(wait);
+            }
+            ushort[] RawPixelData = getFrame();
+            double[] data = new double[pixels];
+
+            lock (lineLock)
+            {
+                if (RawPixelData != null)
+                {
+                    for (int i = 0; i < pixels; ++i)
+                        data[i] = RawPixelData[i + (sampleLine_ * 2048)];
+                }
+            }
+            return data;  
+        }
+
+        public override ushort[] getFrame()
+        {
+            lock(frameLock)
+                return lastFrame;
+        }
+
+        public override string serialNumber
+        {
+            get { return eeprom.serialNumber; }
+        }
+
+        public override float detectorGain
```

Comment:
I think I'm going to leave them like this for now. We do have support for gain and offset for these units, at least looking at Robert's class, so having them like this can serve as a remind to do work on them eventually (I'm focused on minimally viable for the moment)